lib/hanami/application/settings/dotenv_store.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "dry/core/constants"
+
+module Hanami
+  class Application
+    class Settings
+      # Default application settings store.
+      #
+      # Uses dotenv (if available) to load .env files and then loads settings
+      # from ENV.
+      #
+      # @since 2.0.0
+      # @api private
+      class DotenvStore
+        Undefined = Dry::Core::Constants::Undefined
+
+        def fetch(name, default_value = Undefined, &block)
+          name = name.to_s.upcase
+          args = (default_value == Undefined) ? [name] : [name, default_value]
+
+          ENV.fetch(*args, &block)
```

Comment:
@waiting-for-dev It's fine for this PR, but we should not use `ENV` directly. This had problems with Hanami 1, that we should avoid with 2.0. Please have a look at this [article](https://lucaguidi.com/2016/12/27/isolate-global-state/).

Again, not for this PR, but could you follow up with another one fixing this problem? We could either reuse the same Hanami 1 approach, or use `dry-effects`. Thanks 😉

Reply:
Hey @jodosha, I completely agree with you. I ended up doing it in this PR as the change is minimal. Please, take a look, and if you still prefer, I can revert it and do it on another one. I followed the approach in your article so that we don't add `dry-effects` as a dependency until the benefit is greater.

We still depend directly on `ENV` to decide with `.env` files are needed, so at this point, our injection is a bit artificial. We need to update how `Hanami.env` is fetched first, and that definitely belongs to another PR :slightly_smiling_face: